lib/models/src/main/java/org/sagebionetworks/repo/model/NodeDAO.java

```diff
@@ -687,5 +687,12 @@
 	 * @return
 	 */
 	public List<IdAndChecksum> getIdsAndChecksumsForObjects(Long salt, Set<Long> objectIds, Long limit, Long offset);
+	
+	/**
+	 * Get the defining SQL for the given MaterializedView.
+	 * @param id
+	 * @return Optional.empty if the given ID does not belong to a materialized view.
+	 */
+	Optional<String> getMaterializedViewDefiningSql(String id);

```

Comment:
Maybe we should put this is the dao of the materialized view instead (even if it accesses another table)